x/concentrated-liquidity/types/keys.go

```diff
@@ -167,13 +179,13 @@ func KeyAddressPoolIdPositionId(addr sdk.AccAddress, poolId uint64, positionId u
 // KeyAddressAndPoolId returns the prefix key used to create KeyAddressPoolIdPositionId, which only includes addr + pool id.
 // This key can be used to iterate over users positions for a specific pool.
 func KeyAddressAndPoolId(addr sdk.AccAddress, poolId uint64) []byte {
-	return []byte(fmt.Sprintf("%s%s%x%s%d", PositionPrefix, KeySeparator, addr.Bytes(), KeySeparator, poolId))
+	return []byte(fmt.Sprintf("%s%s%x%s%d%s", PositionPrefix, KeySeparator, addr.Bytes(), KeySeparator, poolId, KeySeparator))
 }
 
 // KeyUserPositions returns the prefix key used to create KeyAddressPoolIdPositionId, which only includes the addr.
 // This key can be used to iterate over all positions that a specific address has.
 func KeyUserPositions(addr sdk.AccAddress) []byte {
-	return []byte(fmt.Sprintf("%s%s%x", PositionPrefix, KeySeparator, addr.Bytes()))
+	return []byte(fmt.Sprintf("%s%s%x%s", PositionPrefix, KeySeparator, addr.Bytes(), KeySeparator))
```

Comment:
We make these more constrained, by adding a key separator to the end. This is needed for safety of `KeyUserPositions`, since otherwise you would get other addresses. This is because address lengths are not constant.